src/pycolmap/estimators/covariance.cc

```diff
@@ -0,0 +1,92 @@
+#include "colmap/estimators/covariance.h"
+
+#include "pycolmap/pybind11_extension.h"
+
+#include <pybind11/eigen.h>
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+#include <pybind11/stl_bind.h>
+
+using namespace colmap;
+using namespace pybind11::literals;
+namespace py = pybind11;
+
+void BindCovarianceEstimator(py::module& m) {
+  m.def(
+      "estimate_pose_covariance_from_ba_ceres_backend",
+      [](ceres::Problem* problem,
+         Reconstruction* reconstruction) -> py::object {
+        std::map<image_t, Eigen::MatrixXd> res;
+        if (!EstimatePoseCovarianceCeresBackend(problem, reconstruction, res))
+          return py::none();
+        return py::cast(res);
+      },
+      py::arg("problem"),
+      py::arg("reconstruction"));
+
+  m.def(
+      "estimate_pose_covariance_from_ba",
+      [](ceres::Problem* problem,
+         Reconstruction* reconstruction,
+         double lambda) -> py::object {
+        std::map<image_t, Eigen::MatrixXd> res;
+        if (!EstimatePoseCovariance(problem, reconstruction, res, lambda))
+          return py::none();
+        return py::cast(res);
+      },
+      py::arg("problem"),
+      py::arg("reconstruction"),
+      py::arg("lambda") = 1e-6);
+
+#define REGISTER_ESTIMATOR(NAME)                                               \
```

Comment:
Instead of a macro, can't you instead bind the base class, such that the derived ones do not need to bind inherited attributes twice? https://pybind11.readthedocs.io/en/stable/classes.html#inheritance-and-automatic-downcasting

Reply:
I dont like the idea of binding the base class. It should not be exposed as it is virtual.

Reply:
Why? If you don't bind a constructor for the base class then users won't be able to create any instance of it.

Reply:
Ah OK then. Let me do it later today.